dds/DCPS/SafetyProfileStreams.cpp

```diff
@@ -14,85 +14,67 @@ OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
 namespace OpenDDS {
 namespace DCPS {
 
+String to_dds_string(ACE_CDR::Octet to_convert, bool as_hex)
+{
+  static const int buff_size = 3 + 1; // note +1 for null terminator
+  char buf[buff_size];
+  ACE_OS::snprintf(&buf[0], buff_size, as_hex ? "%02x" : "%d", to_convert);
+  return String(buf);
+}
+
 String to_dds_string(unsigned short to_convert)
 {
-  const char* fmt = "%hu";
-  const int buff_size = 5 + 1; // note +1 for null terminator
+  static const int buff_size = 5 + 1; // note +1 for null terminator
   char buf[buff_size];
-  ACE_OS::snprintf(&buf[0], buff_size, fmt, to_convert);
+  ACE_OS::snprintf(&buf[0], buff_size, "%hu", to_convert);
   return String(buf);
 }
 
 String to_dds_string(int to_convert)
 {
-  const char* fmt = "%d";
-  const int buff_size = 20 + 1; // note +1 for null terminator
+  static const int buff_size = 20 + 1; // note +1 for null terminator
   char buf[buff_size];
-  ACE_OS::snprintf(&buf[0], buff_size, fmt, to_convert);
+  ACE_OS::snprintf(&buf[0], buff_size, "%d", to_convert);
   return String(buf);
 }
 
 String to_dds_string(unsigned int to_convert, bool as_hex)
 {
-  const char* fmt;
-  if (as_hex) {
-    fmt = "%02x";
-    const int buff_size = 3; // note +1 for null terminator
-    char buf[buff_size];
-    ACE_OS::snprintf(&buf[0], buff_size, fmt, to_convert);
-    return String(buf);
-  } else {
-    fmt = "%u";
-    const int buff_size = 20 + 1; // note +1 for null terminator
-    char buf[buff_size];
-    ACE_OS::snprintf(&buf[0], buff_size, fmt, to_convert);
-    return String(buf);
-  }
+  static const int buff_size = 20 + 1; // note +1 for null terminator
+  char buf[buff_size];
+  ACE_OS::snprintf(&buf[0], buff_size, as_hex ? "%x" : "%u", to_convert);
```

Comment:
Did you intend to now use `%08x`?

Reply:
No, I meant to come back to it after I had a better idea of what the formats should be.